docs/04-reference/winglang-spec.md

```diff
@@ -2062,11 +2064,52 @@ let bucket = cdk.aws_s3.Bucket(
 );
 ```
 
-### 5.2 Internal Libraries
+### 5.1.2 Internal Libraries
 
 Wing libraries themselves are JSII modules. They can be used in all other JSII
 supported languages.
 
+## 5.2 JavaScript
+
+The `extern "file.js"` modifier can be used on method declarations (in classes and resources) to indicate that a method is backed by an implementation imported from a JavaScript file.
+
+In the following example, the static inflight method `make_id` is implemented
+in `helper.js`:
+
+```js
+// task-list.w
+resource TaskList {
+  // ...
+
+  inflight add_task(title: str) {
+    let id = TaskList.uuid();
+    this.bucket.put(id, title);
+  }
+
+  extern "./helpers.js" static inflight make_id(): str;
+} 
+
+// helpers.js
+const uuid = require('uuid');
+
+exports.make_id = function() {
+  return uuid.v6();
+};
```

Comment:
Can we add a paragraph to the spec describing what the contract is? like "Given a method of name X, the compiler will map the method to the JavaScript export with the matching name"

If I'm understanding the mapping you're proposing, then if I do this:

```
resource A {
  extern "./helper.js" static inflight make_id(): str;
}

resource B {
  extern "./helper.js" inflight make_id(): str;
}
```

They will be associated with the same JavaScript functions just because they have the same method name - do I understand correctly?

Also, does this mean if "./helpers.js" has a method with a camelCase name then I need to use the camelCase name in Wing? Or will Wing auto-convert it? (food for thought)

Reply:
> Can we add a sentence or two to the spec describing what the contract is? like "Given a method of name X, the compiler will map the method to the JavaScript export with the matching name"

Yes.

> If I'm understanding the mapping you're proposing, then if I do this:

Yes, they will be associated with the same export.

> Also, does this mean if "./helpers.js" has a method with a camelCase name then I need to use the camelCase name in Wing? Or will Wing auto-convert it? (food for thought)

Added - no case conversion here. Generally, case conversion should be only one-way (from JSII to Wing). Since case conversion is lossy, the direction Wing->JSII should just be based on the original symbol and  not reverse conversion (see [#1049](https://github.com/winglang/wing/issues/1049)).